types/ws/ws-tests.ts

```diff
@@ -314,3 +314,17 @@ function f() {
         }
     });
 }
+
+declare module 'ws' {
+    interface WebSocket {
+        id?: string;
+    }
+}
+
+{
+    const ws: wslib.WebSocket = new wslib.WebSocket('ws://www.host.com/path');
+
+    // $ExpectType string | undefined
+    ws.id;
+    ws.id = 'foo';
+}
```

Comment:
```suggestion
```

Module augmentation is the feature in `TypeScript`. I do not think that there is a need to test explicitly, like

```typescript
declare module 'ws' {
    interface WebSocket {
```

Reply:
={

Reply:
=I'd.play

Reply:
> but if it's ever changed back to a type assignment, then that test would fail.

Agree, but the owners of the package would not allow that (I think) because of what we discuss here. I would just leave the test here to explain the motivation of this PR (as an example of usage).

Reply:
I this test is worth keeping in case someone tries to change this back to a `type` assignment. A comment linking to this PR would be helpful but not required.

@cwadrupldijjit, do you want to merge this PR first, as it is now? I'm happy to approve this now and deal with conflicts in the other two PRs

Reply:
I'm fine with that.  I don't mind which order things are merged together.  I'm also fine with helping to deal with the merge conflicts.

Reply:
> this test is worth keeping in case someone tries to change this back to a type assignment.

Let's keep it if this is the concern. Is something else blocking this PR from merging?

Reply:
Since we want to keep the test to prevent changing `interface` to `type`, we should add a test for the `Server` class as well (for consistency).

@cwadrupldijjit Could you consider adding it? Something like

```typescript
declare module 'ws' {
    interface Server {
        id?: string;
    }
}
{
    const ws = new wslib.Server({});
    // $ExpectType string | undefined
    ws.id;
    ws.id = 'foo';
}
```

should be enough.

Reply:
@vansergen, that's a great point.  Let me add that real quick.

Reply:
I've adjusted the test to include a server augmentation, too.  Could you, @vansergen and @k-yle, review it and see if it looks good to you?